flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -536,6 +541,68 @@ public int compare(JobStatusMessage o1, JobStatusMessage o2) {
 	}
 
 	/**
+	 * Executes the STOP action.
+	 * 
+	 * @param args Command line arguments for the stop action.
+	 */
+	protected int stop(String[] args) {
+		LOG.info("Running 'stop' command.");
+
+		StopOptions options;
+		try {
+			options = CliFrontendParser.parseStopCommand(args);
+		} catch (CliArgsException e) {
+			return handleArgException(e);
+		} catch (Throwable t) {
+			return handleError(t);
+		}
+
+		// evaluate help flag
+		if (options.isPrintHelp()) {
+			CliFrontendParser.printHelpForStop();
+			return 0;
+		}
+
+		String[] stopArgs = options.getArgs();
+		JobID jobId;
+
+		if (stopArgs.length > 0) {
+			String jobIdString = stopArgs[0];
+			try {
+				jobId = new JobID(StringUtils.hexStringToByte(jobIdString));
+			} catch (Exception e) {
+				LOG.error("Error: The value for the Job ID is not a valid ID.");
+				System.out.println("Error: The value for the Job ID is not a valid ID.");
+				return 1;
+			}
+		} else {
+			LOG.error("Missing JobID in the command line arguments.");
+			System.out.println("Error: Specify a Job ID to stop a job.");
+			return 1;
+		}
+
+		try {
+			ActorGateway jobManager = getJobManagerGateway(options);
+			Future<Object> response = jobManager.ask(new StopJob(jobId), askTimeout);
+
+			try {
+				Await.result(response, askTimeout);
+
+				Object rc = response.value().get().get();
+				if(rc instanceof StoppingFailure) {
+					throw ((StoppingFailure)rc).cause();
+				}
+
+				return 0;
+			} catch (Exception e) {
+				throw new Exception("Stopping the job with ID " + jobId + " failed.", e);
+			}
+		} catch (Throwable t) {
```

Comment:
I did this the same way as `list()`, `info()`,  `cancel()` are implemented. So I cannot answer. But if we don't catch `Throwable` here, the other methods should not catch either.